File: Unit3L1.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-15
// Fileoverview: This program calculates the total cost of a car based on selected features.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	// Base price
	basePrice := 25000.00
	totalCost := basePrice

	// Feature choices
	var addFloorMats string
	var addNavigation string
	var addHeatedSeats string
	var addWarranty string

	reader := bufio.NewReader(os.Stdin)

	// Input feature choices
	fmt.Print("Would you like floor mats for $500? (yes/no) ")
	addFloorMats, _ = reader.ReadString('\n')
	addFloorMats = strings.TrimSpace(addFloorMats)

	fmt.Print("Would you like a navigation system for $1000? (yes/no) ")
	addNavigation, _ = reader.ReadString('\n')
	addNavigation = strings.TrimSpace(addNavigation)

	fmt.Print("Would you like heated leather seats for $500? (yes/no) ")
	addHeatedSeats, _ = reader.ReadString('\n')
	addHeatedSeats = strings.TrimSpace(addHeatedSeats)

	fmt.Print("Would you like a 5-year extended warranty for $2500? (yes/no) ")
	addWarranty, _ = reader.ReadString('\n')
	addWarranty = strings.TrimSpace(addWarranty)

	// Output base
	fmt.Printf("Base Price                              %.2f\n", basePrice)

	// Add features
	if strings.ToLower(addFloorMats) == "yes" {
		fmt.Println("Floor mats                              500.00")
		totalCost += 500
	}

	if strings.ToLower(addNavigation) == "yes" {
		fmt.Println("Navigation system                      1000.00")
		totalCost += 1000
	}

	if strings.ToLower(addHeatedSeats) == "yes" {
		fmt.Println("Heated leather seats                    500.00")
		totalCost += 500
	}

	if strings.ToLower(addWarranty) == "yes" {
		fmt.Println("5-Year extended warranty               2500.00")
		totalCost += 2500
	}

	// Tax
	taxAmount := totalCost * 0.13
	fmt.Printf("13%% Taxes                              %.2f\n", taxAmount)

	// Final total
	finalCost := totalCost + taxAmount
	fmt.Printf("Final cost of car                      %.2f\n", finalCost)

	fmt.Println("\nDone.")
}

File: Unit3L1.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-15
 * @fileoverview This program calculates the total cost of a car based on selected features.
 */

// variables
const basePrice: number = 25000.0;
let totalCost1: number = basePrice;

// ask for features
const addFloorMats: string = prompt("Would you like floor mats for $500? (yes/no)") || "no";
const addNavigation: string = prompt("Would you like a navigation system for $1000? (yes/no)") || "no";
const addHeatedSeats: string = prompt("Would you like heated leather seats for $500? (yes/no)") || "no";
const addWarranty: string = prompt("Would you like a 5-year extended warranty for $2500? (yes/no)") || "no";

// output base price
console.log("Base Price                              " + basePrice.toFixed(2));

// check each feature
if (addFloorMats.toLowerCase() === "yes") {
  console.log("Floor mats                              500.00");
  totalCost1 += 500;
}

if (addNavigation.toLowerCase() === "yes") {
  console.log("Navigation system                      1000.00");
  totalCost1 += 1000;
}

if (addHeatedSeats.toLowerCase() === "yes") {
  console.log("Heated leather seats                    500.00");
  totalCost1 += 500;
}

if (addWarranty.toLowerCase() === "yes") {
  console.log("5-Year extended warranty               2500.00");
  totalCost1 += 2500;
}

// tax calculation
const taxAmount: number = totalCost * 0.13;
console.log("13% Taxes                              " + taxAmount.toFixed(2));

// final cost
const finalCost: number = totalCost + taxAmount;
console.log("Final cost of car                      " + finalCost.toFixed(2));

console.log("\nDone.");
